integration_test/test/recording_endpoint_test.exs

```diff
@@ -121,7 +120,109 @@ defmodule Membrane.RTC.RecordingEndpointTest do
     assert_receive %TrackAdded{endpoint_id: ^video_file_endpoint_id}, @tracks_added_delay
     assert_receive %TrackAdded{endpoint_id: ^audio_file_endpoint_id}, @tracks_added_delay
     assert_receive %TrackRemoved{endpoint_id: ^video_file_endpoint_id}, @tracks_removed_delay
-    assert_receive %TrackRemoved{endpoint_id: ^audio_file_endpoint_id}
+    assert_receive %TrackRemoved{endpoint_id: ^audio_file_endpoint_id}, @tracks_removed_delay
+
+    assert_receive %EndpointRemoved{endpoint_id: ^recording_endpoint_id}
+
+    await_report(report_file_path)
+    assert %{"tracks" => tracks} = report_file_path |> File.read!() |> Jason.decode!()
+
+    {audio_file, _metadata} =
+      Enum.find(tracks, fn {_filename, track} -> track["type"] == "audio" end)
+
+    {video_file, _metadata} =
+      Enum.find(tracks, fn {_filename, track} -> track["type"] == "video" end)
+
+    audio_deserializer =
+      create_audio_deserializer(%{
+        source: Path.join(output_dir, audio_file),
+        output: deserialized_audio_path,
+        owner: self()
+      })
+
+    video_deserializer =
+      create_video_deserializer(%{
+        source: Path.join(output_dir, video_file),
+        output: deserialized_video_path,
+        owner: self()
+      })
+
+    audio_pipeline = Membrane.Testing.Pipeline.start_link_supervised!(spec: audio_deserializer)
+    video_pipeline = Membrane.Testing.Pipeline.start_link_supervised!(spec: video_deserializer)
+
+    assert_end_of_stream(audio_pipeline, :sink)
+    assert_end_of_stream(video_pipeline, :sink)
+
+    assert deserialized_audio_path |> File.read!() |> byte_size() > 0
+    assert deserialized_video_path |> File.read!() |> byte_size() > 0
+  end
+
+  @tag :tmp_dir
+  test "engine doesn't crash when recording endpoint is removed too early",
```

Comment:
Why would it?

Reply:
Before these changes, the engine would throw an exception when handling a subscribe message from an endpoint that didn't exist in the engine state. I should probably change the test name to something like `engine doesn't throw exception when recording endpoint is removed before engine handles endpoint subscribe requests`, but it very long name for a test.